web/src/veterans/VeteranDialog.js

```diff
@@ -370,13 +384,23 @@ class VeteranDialog extends Component {
                 <ConfirmDialog
                     title={'Confirm Veteran Deletion'}
                     prompt={'Delete'}
-                    open={this.state.confirmDialog.open}
+                    open={this.state.confirmDialog === 'delete'}
                     onConfirm={this.handleDeleteConfirmClick}
                     onClose={this.handleDialogClose}
                     suppressCloseOnConfirm
                 >
                     <p>Are you sure you want to delete Veteran '{firstName + ' ' + lastName}'?</p>
                 </ConfirmDialog>
+                <ConfirmDialog
+                    title={'Confirm Card Number Change'}
+                    prompt={'Change'}
+                    open={this.state.confirmDialog === 'changeCardNumber'}
+                    onConfirm={this.handleUpdateConfirmClick}
+                    onClose={this.handleDialogClose}
+                    suppressCloseOnConfirm
+                >
+                    <p>Are you sure you want to change this Veteran's card number? The old card will no longer work.</p>
```

Comment:
Let's add something like, 'Are you sure you want to change this Veteran's card number to 113242?  The previous card, 113241, will no longer function'.